queue_consumer.go

```diff
@@ -92,7 +93,7 @@ func (mf *Consumer) receiveMessages(ctx context.Context, wg *sync.WaitGroup, don
 
 	rcvParams := &sqs.ReceiveMessageInput{
 		QueueUrl:            mf.s.URL,
-		MaxNumberOfMessages: aws.Int64(awsBatchSizeLimit),
+		MaxNumberOfMessages: aws.Int64(mf.MessagesBatchSizeLimit),
```

Comment:
I wonder if we should cap this value at 10, so a misconfigured client will still function, or whether we should rely on the error messages in the logs.

Ideally a process will return an error if the configuration is invalid and we can choose to crash, but with the way this package is coded a caller can't react to this error in code.

It could be that the AWS API already does this for us, I don't know.  My vote would be to make it work "gracefully" and simply limit the value to 10 even if a caller set it to 100.

Reply:
I doubt if AWS doesn't return an error or at least they cap it. I don't think they will return > 10 messages. This type of validation is on AWS to do and return the error.  I don't know why we would duplicate this validation here. Even if we can cap the max number of messages here, the caller can still misconfigure other configs (eg wait time in seconds, etc). We can't account for every possible scenario.

Yea, I agree, the way it is coded doesn't allow the caller to react to the errors, just monitor the logs. Ideally, as you said, we should return an error and let the caller handle it gracefully but this is out of the scope of this PR.